robottelo/hosts.py

```diff
@@ -984,16 +984,21 @@ def __enter__(self):
         """Satellite objects can be used as a context manager to temporarily force everything
         to use the Satellite object's hostname.
         """
-        self.__old_hostname = settings.server.hostname
-        settings.server.hostname = self.hostname
-        configure_nailgun()
-        configure_airgun()
+        self._revert = False
+        # if the hostname is the same, don't make a change
+        if self.hostname != settings.server.hostname:
+            self._revert = True
+            self._old_hostname = settings.server.hostname
+            settings.server.hostname = self.hostname
```

Comment:
Use `set` method here?

Reply:
that was a red herring, it had the same effect :/